http_server.py

```diff
@@ -330,20 +330,46 @@ async def submit_vote_handler(request: web_request.Request) -> web.Response:
 
     user_id = keys.vote_keys[auth_key]["userID"]
     user_name = keys.vote_keys[auth_key]["userName"]
+    user_votes = vote_input["votes"]
 
     week = compo.get_week(False)
 
-    if not "votes" in week:
-        week["votes"] = []
-
     # If user has submitted a vote already, then remove it, so we can
     # replace it with the new one
     for v in week["votes"]:
         if int(v["userID"]) == int(user_id):
             week["votes"].remove(v)
 
+    # Find the user's entry
+    user_entry = None
+
+    for entry in week["entries"]:
+        if entry["entrantName"] == user_name:
```

Comment:
```suggestion
        if entry["discordID"] == user_id:
```
We should be identifying entries by entrant's discord ID, not entrant's discord username